crates/gitbutler-repo/src/rebase.rs

```diff
@@ -225,11 +227,84 @@ fn commit_conflicted_cherry_result<'repository>(
         .context("failed to find commit")
 }
 
+fn extract_conflicted_files(
+    merged_tree_id: gix::Id<'_>,
+    merge_result: gix::merge::tree::Outcome<'_>,
+    treat_as_unresolved: gix::merge::tree::TreatAsUnresolved,
+) -> Result<ConflictEntries> {
+    use gix::index::entry::Stage;
+    let repo = merged_tree_id.repo;
+    let mut index = repo.index_from_tree(&merged_tree_id)?;
+    merge_result.index_changed_after_applying_conflicts(
+        &mut index,
+        treat_as_unresolved,
+        gix::merge::tree::apply_index_entries::RemovalMode::Mark,
+    );
+    let (mut ancestor_entries, mut our_entries, mut their_entries) =
+        (Vec::new(), Vec::new(), Vec::new());
+    let conflicting_entries = index.entries().iter().filter_map(|e| {
+        let stage = e.stage();
+        if stage == gix::index::entry::Stage::Unconflicted {
+            None
+        } else {
+            Some((stage, e))
+        }
+    });
+    for (stage, entry) in conflicting_entries {
+        let storage = match stage {
+            Stage::Unconflicted => {
+                unreachable!("BUG: filtered above to not contain unconflicted entries")
+            }
+            Stage::Base => &mut ancestor_entries,
+            Stage::Ours => &mut our_entries,
+            Stage::Theirs => &mut their_entries,
+        };
```

Comment:
Why filter above, when you could just put a `continue;` on line 256?

Reply:
A great catch, I changed it and it's so much easier to read.

Reply:
Hi, @Byron. GitHub is saying that this has changed since I last viewed it, but I'm not actually seeing the changes. Is GitHub being broken or have your amendments not been pushed? Sorry for the bother.

Reply:
There is definitely room to innovate in the PR workflow as well :). And I have no clue how to most easily get to the updated version.

Here is the commit that made the change, there is only one: c64b84032ea963d188ba414a1fb40900ccbb9489